beastling/clocks/baseclock.py

```diff
@@ -1,24 +1,24 @@
 from beastling.util import xml
+from beastling.util.misc import FromOptions
 
 
-class BaseClock(object):
+class BaseClock(FromOptions):
+    __type__ = None
+    __distribution__ = None
```

Comment:
These two `__`-attributes relate a clock implementation to the corresponding config options, thus following the "principle of locality" - rather than having a hard-coded lookup table in `Configuration`. The implementation lookup now happens in `sections.Clock.get_clock`, iterating over all subclasses of `BaseClock`.